src/test/java/org/javarosa/core/util/GeoDistanceTest.java

```diff
@@ -112,4 +114,60 @@ public void distance_isComputedForGeoshape() throws IOException {
         assertThat(Double.parseDouble(scenario.answerOf("/data/distance").getDisplayText()),
             closeTo(NINETY_DEGREES_ON_EQUATOR_KM * 2, 1e-7));
     }
+
+    @Test
+    public void distance_isComputedForString() throws IOException {
+        Scenario scenario = Scenario.init("string distance", html(
+            head(
+                title("String distance"),
+                model(
+                    mainInstance(t("data id=\"string-distance\"",
+                        t("point1", "38.253094215699576 21.756382658677467 0 0"),
+                        t("point2", "38.25021274773806 21.756382658677467 0 0"),
+                        t("point3", "38.25007793942195 21.763892843919166 0 0"),
+                        t("point4", "38.25290886154963 21.763935759263404 0 0"),
+                        t("point5", "38.25146813817506 21.758421137528785 0 0"),
+                        t("concat"),
+                        t("distance")
+                    )),
+                    bind("/data/point1").type("geopoint"),
+                    bind("/data/point2").type("geopoint"),
+                    bind("/data/point3").type("geopoint"),
+                    bind("/data/point4").type("geopoint"),
+                    bind("/data/point5").type("geopoint"),
+                    bind("/data/concat").type("string").calculate("concat(/data/point1, ';', /data/point2, ';', /data/point3, ';', /data/point4, ';', /data/point5)"),
+                    bind("/data/distance").type("decimal").calculate("distance(/data/concat)")
+                )
+            ),
+            body(
+                input("/data/point1")
```

Comment:
Does this need to have an input?

Reply:
Removed the entire body for this one and the corresponding area test. Thanks!